### PR TITLE
openbsd: export `struct sigcontext` for x86_64

### PR DESCRIPTION
**Pull request by semarie**
based on initial work from @devnexen (see #2189)

tested on OpenBSD (7.0-beta) with rustc 1.54.0

### CODE REVIEW

File: src/unix/bsd/netbsdlike/openbsd/x86_64.rs [hunk 1]
```diff
@@ -3,6 +3,110 @@ use PT_FIRSTMACH;
...
+        pub sc_es: ::c_long,
+        pub sc_ds: ::c_long,
+        pub sc_trapno: ::c_long,
+        pub sc_err: ::c_long,
+        pub sc_rip: ::c_long,
+        pub sc_cs: ::c_long,
+        pub sc_rflags: ::c_long,
+        pub sc_rsp: ::c_long,
+        pub sc_ss: ::c_long,
+        sc_fpstate: *mut fxsave64,
```
Comment (JohnTitor) [inline src/unix/bsd/netbsdlike/openbsd/x86_64.rs:36]: Should this be private?

File: src/unix/bsd/netbsdlike/openbsd/x86_64.rs [hunk 2]
```diff
@@ -3,6 +3,110 @@ use PT_FIRSTMACH;
...
+        pub fx_xmm: [[u64; 2]; 16],
+        __fx_unused3: [u8; 96],
+    }
+}
+
+cfg_if! {
+    if #[cfg(feature = "extra_traits")] {
+        impl PartialEq for fxsave64 {
+            fn eq(&self, other: &fxsave64) -> bool {
+                return {self.fx_fcw} == {other.fx_fcw} &&
```
Comment (JohnTitor) [inline src/unix/bsd/netbsdlike/openbsd/x86_64.rs:65]: Other places don't use curly brackets, let's get rid of them for consistency.

File: src/unix/bsd/netbsdlike/openbsd/x86_64.rs (hunk 1 above)
Comment (semarie) [reply src/unix/bsd/netbsdlike/openbsd/x86_64.rs:36]: hum, no.

File: src/unix/bsd/netbsdlike/openbsd/x86_64.rs (hunk 2 above)
Comment (semarie) [reply src/unix/bsd/netbsdlike/openbsd/x86_64.rs:65]: curly brackets are required for `fxsave64` fields: it creates a temporary storage, the value is copied from `fxsave64` field, and operation is done on this storage. due to the fact `fxsave64` is packed, direct access of a field implies an unaligned access, and rustc complains with `error: reference to packed field is unaligned` (see https://github.com/rust-lang/rust/issues/82523)

Comment (semarie) [reply src/unix/bsd/netbsdlike/openbsd/x86_64.rs:65]: in libc, `epoll_event` in linux-like is also `packed`. the field access is done using explicit `let` (for copying the value in aligned storage) but there is only two fields in `epoll_event`. it is why I used temporary storage with curly brackets.

Comment (JohnTitor) [reply src/unix/bsd/netbsdlike/openbsd/x86_64.rs:65]: Ah, I see! Then could you add a comment to explain it? r=me once it's added and commits are squashed.

Comment (semarie) [reply src/unix/bsd/netbsdlike/openbsd/x86_64.rs:67]: done. thanks.

**Review by JohnTitor — APPROVED** [review]

**Comment by JohnTitor** [comment]
Thanks!
@bors r+

**Pull request merged**